### DISCUSSION TITLE
How to pass parameter from widget to generated riverpod provider

### DISCUSSION THREAD

**Discussion by teranpeterson** [rrousselGit/riverpod#2642] [Q&A] [answered]
I have an AccountList page. When you click on an account, an AccountDetails page is shown. That page takes, as a parameter, the account number. How do I pass that account number to my AccountDetailsPageController? Right now I am doing the following:

AccountDetailsController:
```dart
@riverpod
class AccountDetailsPageController extends _$AccountDetailsPageController {
  @override
  FutureOr<AccountDetailsPageState> build(String accountNumber) async {
    final account = await ref.read(accountsRepositoryProvider).getAccount(accountNumber);
    
    return AccountDetailsPageState(account: account);
  }
}
```

AccountDetailsPage:
```dart
class AccountDetailsPage extends ConsumerWidget {
  const AccountDetailsPage({super.key, required this.accountNumber});

  final String accountNumber;

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    // Demos of initializing the controller
    ref.read(accountDetailsPageControllerProvider(accountNumber).notifier);
    ref.watch(accountDetailsPageControllerProvider(accountNumber));
    ref.listen(accountDetailsPageControllerProvider(accountNumber), ...);
    ...
  }
}
```

Which works, but I get the following linter warning: `Notifiers should not have public properties/getters. Instead, all their public API should be exposed through the `state` property.`

Can someone point me in the right direction?

**Comment by rrousselGit** [comment]
> Which works, but I get the following linter warning: `Notifiers should not have public properties/getters. Instead, all their public API should be exposed through the `state` property.`

Where do you get that warning?

The code you gave shouldn't trigger the warning. Either there's something you didn't share or there's a bug somewhere.

**Reply by teranpeterson** [reply]
The specific linter error comes from the generated file account_details_page_controller.g.dart on the `late final` declarations: 

```dart
abstract class _$AccountDetailsPageController
    extends BuildlessAutoDisposeAsyncNotifier<AccountDetailsPageState> {
  late final String accountNumber;

  FutureOr<AccountDetailsPageState> build(
    String accountNumber,
  );
}
```

I have the following linters:

```yaml
flutter_lints: ^2.0.0
custom_lint: ^0.3.4
riverpod_lint: ^1.3.1
```

**Reply by rrousselGit** [reply, marked as answer]
Looks like a bug. I've made an issue about it. Will fix.

For now I'd recommend disabling the analysis of generated files.
